src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -596,6 +600,9 @@ public static void FoundCertWithPrivateKey(this ILogger<HttpsConnectionMiddlewar
     [LoggerMessage(8, LogLevel.Debug, "Failed to open certificate store {StoreName}.", EventName = "FailToOpenStore")]
     public static partial void FailedToOpenStore(this ILogger<HttpsConnectionMiddleware> logger, string? storeName, Exception exception);
 
+    [LoggerMessage(9, LogLevel.Information, "Certificate with thumbprint {Thumbprint} lacks the subjectAlternativeName (SAN) extension and may not be accepted by browsers.", EventName = "NoSubjectAlternativeName")]
```

Comment:
I went with `Information`, rather than `Warning`, because this server might not receive auth requests from browsers (i.e. it might not actually matter that the extension is missing.).

Reply:
Info is fine. I think we need to provide more detail about why browsers might not accept it. Is there a good doc or article that we could link to?

> Certificate with thumbprint {Thumbprint} lacks the subjectAlternativeName (SAN) extension. Some browsers require a SAN and will not accept this certificate. For more information, see https://aka.ms/aspnet/cert-san.

Reply:
I linked an article in the description, but we don't own it.

Reply:
I found this, but it seems to be about running a CA (and seems nervous about using SANs at all?).
https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/ff625722(v=ws.10)?redirectedfrom=MSDN#BKMK_Security

Reply:
@vcsjones Is this (i.e. specifying a subject alternative name) even a sensible thing to warn about / recommend?  Maybe we should only do it for self-signed certs or when the common name is `localhost`?

Reply:
Thanks!

Reply:
Even our internal cert issues implemented SANs 10 years+ ago (Because muggins here wrote the request UI).

But whilst the cab forum rules are meant to be "Thou shalt not" there are moments when CAs forget themselves, due to bugs, so a information message is still a good idea.

Reply:
I don't suppose either @vcsjones or @blowdart is aware of a page I could link to?  (There are some useful pages online, but I understand aka.ms links must point to 1P properties.)

Reply:
I don't know of anything 1st party